drivers/regulator/regulator_fixed.c

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2019-2020 Peter Bigot Consulting, LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT regulator_fixed
+
+#include <drivers/regulator.h>
+#include <drivers/gpio.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(regulator_fixed, CONFIG_REGULATOR_LOG_LEVEL);
+
+#define OPTION_ALWAYS_ON_POS 0
+#define OPTION_ALWAYS_ON BIT(OPTION_ALWAYS_ON_POS)
+#define OPTION_BOOT_ON_POS 1
+#define OPTION_BOOT_ON BIT(OPTION_BOOT_ON_POS)
+
+struct driver_config {
+	const char *gpio_name;
+	uint32_t delay_us;
+	gpio_pin_t gpio_pin;
+	gpio_dt_flags_t gpio_flags;
+	uint8_t options;
+};
+
+struct driver_data {
+	struct device *dev;
+	struct device *gpio;
+	struct onoff_manager mgr;
+};
+
+/* Declaration here, definition after members defined. */
+static const struct regulator_driver_api api;
+
+static void start(struct onoff_manager *mgr,
+		  onoff_notify_fn notify)
+{
+	const struct driver_data *data =
+		CONTAINER_OF(mgr, struct driver_data, mgr);
+	const struct driver_config *cfg = data->dev->config_info;
+	int rc = gpio_pin_set(data->gpio, cfg->gpio_pin, true);
+
+	if (rc == -EWOULDBLOCK) {
+		/* TODO: delegate completion to a work queue. */
+	} else if ((rc >= 0) && (cfg->delay_us > 0)) {
+		if (k_is_in_isr()
+		    || k_is_pre_kernel()
+		    || (k_us_to_ticks_floor32(cfg->delay_us) == 0)) {
```

Comment:
What is the reason for the last check, i.e. `k_us_to_ticks_floor32(cfg->delay_us) == 0)`?

Reply:
If the delay is less than one tick then `k_busy_wait` is required to avoid waiting longer than necessary, because sleep durations are always a multiple of ticks.  If it's at least one tick then `k_usleep()` would save power.

This area needs rework for other reasons.